dbms/src/Interpreters/Settings.h

```diff
@@ -366,7 +366,10 @@ struct Settings
     M(SettingUInt64, manual_compact_max_concurrency, 10, "Max concurrent tasks. It should be larger than pool size.")                                                                                                                   \
     M(SettingUInt64, manual_compact_more_until_ms, 60000, "Continuously compact more segments until reaching specified elapsed time. If 0 is specified, only one segment will be compacted each round.")                                \
                                                                                                                                                                                                                                         \
-    M(SettingBool, enable_planner, true, "Enable planner")
+    M(SettingBool, enable_planner, true, "Enable planner")                                                                                                                                                                              \
+    M(SettingBool, enable_pipeline, false, "Enable pipeline model")                                                                                                                                                                     \
+    M(SettingInt64, pipeline_task_executor_threads, -1, "The thread num of task executor. -1 means using hardware_concurrency.")                                                                                                        \
```

Comment:
For the other settings, `0` meansing using hardware concurrency, why use `-1` here?

Reply:
ohhh..I will replace by `0`

Reply:
updated.